### PR TITLE
Task4

### PR DESCRIPTION
**Pull request by grnkvch**
1. Task: https://github.com/rolling-scopes-school/nodejs-aws-tasks/blob/main/task4-rds/task.md
> Information about setup and testing see README.md
2. Links:
> BACKEND
> https://1ey1lyj9ng.execute-api.eu-west-1.amazonaws.com/dev/product/b823ab27-61f0-4205-9123-87a8480f0b4a  
> https://1ey1lyj9ng.execute-api.eu-west-1.amazonaws.com/dev/product/
>
>SWAGGER https://app.swaggerhub.com/apis-docs/grnkvh/product-service/0.0.1

> FRONTEND
> PR: https://github.com/grnkvch/nodejs-aws-fe/pull/3
> https://dszl4pcvftgd3.cloudfront.net/

3. Deadline 08.11.2020 23:59
4. Score: 8/9
  - [x] Task 4.1 is implemented
  - [x] TASK 4.2 is implemented lambda links are provided and returns data
  - [x] TASK 4.3 is implemented lambda links are provided and products is stored in DB (call TASK 4.2 to see the product)
  - [x] Your own Frontend application is integrated with product service (/products API) and products from product-service are represented on Frontend. Link to a working Front-End application is provided for cross-check reviewer.
  - [x] POST/products lambda functions returns error 400 status code if product data is invalid
  - [x] All lambdas return error 500 status code on any error (DB connection, any unhandled error in code)
  - [x] All lambdas do console.log for each incoming requests and their arguments
  - [ ] Transaction based creation of product (in case stock creation is failed then related to this stock product is not created and not ready to be used by the end user and vice versa)

### CODE REVIEW

File: product-service/ddl.sql [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    description text,
+    price integer,
+    image text
+);
+
+create table stocks (
+	id serial primary key,
+	product_id uuid UNIQUE,
+	count  integer,
+	foreign key ("product_id") references "products" ("id")
```
Comment (valakar) [inline product-service/ddl.sql:13]: Not to current task: Probably you will not be able to delete product row with respected stock available (you will need to add **on delete**)

**Comment by valakar** [comment]
Looks good!

Found only one issue 

> POST/products lambda functions returns error 400 status code if product data is invalid

Tried to send post without required field and received 500 instead of 400:
image

File: product-service/ddl.sql (hunk 1 above)
Comment (grnkvch) [reply product-service/ddl.sql:13]: Sure, i'm going to improve it in future

**Comment by grnkvch** [comment]
> Looks good!
> 
> Found only one issue
> 
> > POST/products lambda functions returns error 400 status code if product data is invalid
> 
> Tried to send post without required field and received 500 instead of 400:
> image

As I can see, you put invalid JSON to body, I don't validate body, but validate field's types inside body.
Hope I will fix it in further task

**Comment by valakar** [comment]
> > Looks good!
> > Found only one issue
> > > POST/products lambda functions returns error 400 status code if product data is invalid
> > 
> > 
> > Tried to send post without required field and received 500 instead of 400:
> > image
> 
> As I can see, you put invalid JSON to body, I don't validate body, but validate field's types inside body.
> Hope I will fix it in further task

Yeah, you are right

**Pull request merged**